### PR TITLE
Fix the DataContainer type hint issue

### LINKED ISSUE #184: Form validation leads to data container type errors in save_callbacks for tl_calendar_events
Hello,

I recently used the Form class which works great so far. Only one thing isn't quite perfect. When using the validate() function, the save_callback of the field endTime (data container tl_calendar_events) tl_calendar_events::setEmptyEndTime() is called out of the Form class.

Unfortunately, this callback has a type hin which is `DataContainer $dc` and the Form class passes in a stdClass. Therefor I get type hint errors.

https://github.com/codefog/contao-haste/blob/main/src/Form/Form.php#L373-L380

I suspect that the reason for this is that by creating a DC_Table instance, callbacks would be called without intention (onload_callback e.g.).

Is there any way to cope with that problem?

Thanks in advance!

bye Defcon0

### PR DESCRIPTION
**Pull request by qzminski**
Closes #184.

### CODE REVIEW

File: src/Form/Form.php [hunk 1]
```diff
@@ -421,6 +406,45 @@ static function (mixed $value, Widget $widget, self $form) use ($fieldConfig, $f
...
+            private ?Model $model;
+
+            public function __construct($model, $value, $field, $name)
+            {
+                $this->id = $model ? $model->id : 0;
+                $this->table = $model ? $model::getTable() : '';
+                $this->value = $value;
+                $this->field = $field;
+                $this->inputName = $name;
+                $this->activeRecord = $model; // BC
```
Comment (Toflar) [inline src/Form/Form.php:424]: `activeRecord` is not BC, it just has a more clear meaning in Contao 5: `getCurrentRecord()` should always return the record as it is currently stored in the DB while `activeRecord` contains the unsafed changes. 

So I think your logic currently represents the same values for both cases which would be wrong afaics.

Comment (Defcon0) [reply src/Form/Form.php:424]: OK, then I'll wait with the testing til the feedback.

Comment (qzminski) [reply src/Form/Form.php:422]: Changed in 2f16ebfc. I added the BC comment because `activeRecord` is deprecated as of Contao 5.0.

**Review by Toflar — APPROVED** [review]
LGTM, can be tested.

**Comment by Defcon0** [comment]
Works 4 me :-) Thanks!

**Comment by qzminski** [comment]
Released as 5.0.5, thanks for input @Defcon0 !

**Pull request merged**